src/host_interface/virtio_console.c

```diff
@@ -185,20 +185,19 @@ static int console_enqueue(
         {
             ret = readv(vcd->in_console_fd, iov, req->buf_count);
         } while (ret == -1 && errno == EINTR);
-
-        if (ret < 0)
-            if (errno != EAGAIN)
-                sgxlkl_host_warn(
-                    "%s: read failed: ret:%d err:%d\n", __func__, ret, errno);
     }
     else
     {
-        ret = writev(vcd->out_console_fd, iov, req->buf_count);
+        do
+        {
+            ret = writev(vcd->out_console_fd, iov, req->buf_count);
+        } while (ret == -1 && errno == EINTR);
     }
 
-    virtio_req_complete(req, ret);
-    // TODO https://github.com/lsds/sgx-lkl/issues/374
-    return 0;
+    ssize_t bytes = ret <= 0 ? 0 : ret;
+    virtio_req_complete(req, bytes);
```

Comment:
We are calling this (and returning success), even if `readv` or `writev` didn't process all of the bytes. Is that correct?

Reply:
Correct as in, is that what we are doing? Yes
Correct as in, the right thing to do? I'm not sure. From my fairly limited understanding from reading the descriptions in headers, I think it is fine.